### PR TITLE
Update/retrofit

### PR DESCRIPTION
**Pull request by mariobodemann**
This PR will 

update okhttp from `2.5.0` to `3.2.0`
updated retrofit from `1.9.0` to `2.0.1` and 
updated rxjava from `1.0.14` to `1.1.2`

together with additional code changes, to make the new libraries run

### CODE REVIEW

File: src/main/java/com/contentful/java/cda/CDAClient.java [hunk 1]
```diff
@@ -228,7 +266,13 @@ public CDASpace fetchSpace() {
     return Observable.just(contentType);
   }
 
-  /** Returns a {@link CDAClient} builder. */
+  String createUserAgent() {
+    return String.format("contentful.java/%s", Util.getProperty("version.name"));
```
Comment (neonichu) [inline src/main/java/com/contentful/java/cda/CDAClient.java]: Might be interesting to add some more information here, e.g. the iOS SDK also logs the OS version.

File: src/test/java/com/contentful/java/cda/CallbackTest.java [hunk 2]
```diff
@@ -21,15 +22,14 @@ public void fetchEntryAsync() throws Exception {
   }
 
   @Test
-  @SuppressWarnings("ThrowableResultOfMethodCallIgnored")
   public void onFailure() throws Exception {
     server.enqueue(new MockResponse().setStatus("404"));
-    TestCallback<CDASpace> callback = client.fetchSpace(new TestCallback<CDASpace>()).await();
-    assertThat(callback.error()).isNotNull();
```
Comment (neonichu) [inline src/test/java/com/contentful/java/cda/CallbackTest.java]: Hm, what's the reasoning for no longer doing these assertions?

**Comment by neonichu** [comment]
LGTM once we have discussed the two comments :ok_hand:

File: src/main/java/com/contentful/java/cda/CDAClient.java (hunk 1 above)
Comment (mariobodemann) [inline src/main/java/com/contentful/java/cda/CDAClient.java]: Sounds like a good idea. I'll take a look on how to determine the current ios version ... ;)

File: src/test/java/com/contentful/java/cda/CallbackTest.java (hunk 2 above)
Comment (mariobodemann) [inline src/test/java/com/contentful/java/cda/CallbackTest.java]: We don't have a callback any more, so we can't make assumptions on it.

**Pull request merged**